python-test-samples/integrated-application-test-kit/README.md

```diff
@@ -0,0 +1,112 @@
+cd ./plugins/plugin_tester && sam deploy --no-confirm-changeset --no-fail-on-empty-changeset
+```
+
+### Run the prevalidate integration tests
+
+The prevalidate plugin is a sample plugin that acts on the [prevalidate event](./events/1-prevalidate-event.md) emitted by the platform. These integration tests use the AWS SDK to query the Amazon CloudFormation API directly and poll the plugin tester workflow.
+
+Deploy the [test-prevalidate-plugin](./plugins/1-prevalidate-plugins/test-prevalidate-plugin/)
+
+```bash
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && sam build --use-container
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && sam deploy --no-confirm-changeset --no-fail-on-empty-changeset
+```
+
+Run the [test-prevalidate-plugin](./plugins/1-prevalidate-plugins/test-prevalidate-plugin/) integration tests
+
+Replace the `AWS_REGION` parameter with the region you deployed to.
+
+```bash
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && npm install
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 npm test
+```
+
+### Run the postvalidate integration tests
+
+The postvalidate plugin is a sample plugin that acts on the [postvalidate event](./events/2-postvalidate-event.md) emitted by the platform. These integration tests use the Integrated Application Test Kit to find cloud-hosted resources and listen for activity.
+
+Deploy the [python-minimal-plugin](./plugins/2-postvalidate-plugins/python-minimal-plugin/)
+
+```bash
+cd ./plugins/2-postvalidate-plugins/python-minimal-plugin && sam build --use-container
+cd ./plugins/2-postvalidate-plugins/python-minimal-plugin && sam deploy --no-confirm-changeset --no-fail-on-empty-changeset
+```
+
+Run the [python-minimal-plugin](./plugins/2-postvalidate-plugins/python-minimal-plugin/) integration tests
+
+Replace the `AWS_REGION` parameter with the region you deployed to.
+
+```bash
+cd ./plugins/2-postvalidate-plugins/python-minimal-plugin/tests/integration
+PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 python3 -m unittest ./test_by_polling.py
+PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 python3 -m unittest ./test_by_waiting.py
+PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 python3 -m unittest ./test_by_polling_with_registry.py
+PLUGIN_TESTER_STACK_NAME=video-plugin-tester AWS_REGION=us-west-2 python3 -m unittest ./test_trace.py
+```
+
+### Clean up
+
+Delete the [python-minimal-plugin](./plugins/2-postvalidate-plugins/python-minimal-plugin/)
+
+```bash
+cd ./plugins/2-postvalidate-plugins/python-minimal-plugin && sam delete
+```
+
+Delete the [test-prevalidate-plugin](./plugins/1-prevalidate-plugins/test-prevalidate-plugin/)
+
+```bash
+cd ./plugins/1-prevalidate-plugins/test-prevalidate-plugin && sam delete
+```
+
+Delete the [plugin tester](./plugins/plugin_tester/)
+
+```bash
+cd ./plugins/plugin_tester && sam delete
+```
```

Comment:
To learn more about testing serverless applications go to [serverlessland.com/testing](serverlessland.com/testing).